pkg/cmd/account/helper/helper.go

```diff
@@ -0,0 +1,38 @@
+package helper
+
+import (
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/environments"
+	"github.com/briandowns/spinner"
+)
+
+// ResolveNamesOrId takes in an array of names and trys to find an exact match.
+// If a match is found it will return its corresponding ID. If no match is found
+// it will return the name as is, in assumption it is an ID.
+func ResolveEnvironmentNames(envs []string, octopus *client.Client, spinner *spinner.Spinner) ([]string, error) {
+	spinner.Start()
+	envIds := make([]string, 0, len(envs))
+loop:
+	for _, envName := range envs {
+		matches, err := octopus.Environments.Get(environments.EnvironmentsQuery{
+			Name: envName,
+		})
+		if err != nil {
+			return nil, err
+		}
+		allMatches, err := matches.GetAllPages(octopus.Environments.GetClient())
+		if err != nil {
+			spinner.Stop()
+			return nil, err
+		}
+		for _, match := range allMatches {
+			if envName == match.Name {
+				envIds = append(envIds, match.ID)
+				continue loop
```

Comment:
I had to think pretty hard to work out the logic here...

Perhaps it would be simpler just to do an ordinary loop with a break (and no continue-outer) and then just put an if statement at the end which says `if(len(envIds) == 0) { envIds = envIds.Append(envName) }`